Dockerfile

```diff
@@ -39,11 +45,82 @@ RUN mkdir -p $JENKINS_HOME/init.groovy.d \
     && mkdir $JENKINS_HOME/utils \
     && mkdir $JENKINS_HOME/git \
     && mkdir -p /var/log/jenkins
-COPY src/main/groovy/*.groovy $JENKINS_HOME/init.groovy.d/
 COPY plugins $JENKINS_HOME/plugins/
 COPY utils/ $JENKINS_HOME/utils/
 COPY ${CONFIG_PATH} $JENKINS_HOME/init-configs/
 
+# -- test shard #1
+# -- copy the scripts used to configure a Jenkins container for
+# -- running the tests specifed with the environment variable
+# -- 'TEST_SHARD=shard_1'
+
+FROM base as shard_1
+ARG user=jenkins
+ARG group=jenkins
+ARG uid=1000
+ARG gid=1000
+COPY src/main/groovy/1addJarsToClasspath.groovy \
```

Comment:
I think it might be worth pulling out all of the duplicate files between shards and adding it to the "base" section of this dockerfile. That way it will be very obvious which files are unique in each shard, rather than having to dig and do a painful comparison.